### PR TITLE
feat: exclude users from rankings that have 0 points

### PR DESCRIPTION
**Pull request by jowparks**
## Summary
1. Excludes users from rank list if they have 0 points
2. When user is searched for, return users even if they have no points
3. Lifetime points should return 0 for user (without rank query) if user has no points.

## Testing Plan
Unittest

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: src/users/users.service.ts [hunk 1]
```diff
@@ -269,7 +271,7 @@ export class UsersService {
             ) AS rank
           FROM
             users
-          JOIN
+          INNER JOIN
```
Comment (rohanjadvani) [inline src/users/users.service.ts:274]: If I inner join with `user_latest_events`, will the search show a user if they don't have points? That'd seem a little confusing to me if I want to look at my profile but haven't participated yet.

Comment (jowparks) [reply src/users/users.service.ts:274]: Yup you are right, users would be filtered out here if they have no points. So should we only filter them from list result?

File: src/events/events.service.ts [hunk 2]
```diff
@@ -702,6 +702,15 @@ export class EventsService {
       .map((e) => e.toLowerCase() + '_last_occurred_at')
       .join(', ');
 
+    // return early if user doesn't have any points
+    const userPoints = await this.userPointsService.findOrThrow(user.id);
+    if (userPoints.total_points === 0) {
+      return {
+        rank: 0,
```
Comment (NullSoldier) [inline src/events/events.service.ts:709]: We should return null for the rank here

File: src/events/events.service.ts [hunk 3]
```diff
@@ -691,7 +691,27 @@ export class EventsService {
...
     events: EventType[],
-  ): Promise<{ points: number; count: number; rank: number }> {
+  ): Promise<{
+    points: number | null;
+    count: number | null;
+    rank: number | null;
+  }> {
+    // return early if user doesn't have any points
+    const userPoints = await this.userPointsService.findOrThrow(user.id);
+    const filteredPoints = events
```
Comment (jowparks) [inline src/events/events.service.ts:701]: yuck, but has to be done this way based on duck typing

File: src/events/events.service.ts [hunk 4]
```diff
@@ -747,9 +771,9 @@ export class EventsService {
     }
 
     return {
-      rank: rank[0].rank,
-      points: rank[0].points,
-      count: rank[0].count,
+      rank: rank[0].rank || null,
+      points: rank[0].points || null,
+      count: rank[0].count || null,
```
Comment (NullSoldier) [inline src/events/events.service.ts:776]: Remove these || null

File: src/users/users.controller.spec.ts [hunk 5]
```diff
@@ -349,6 +349,40 @@ describe('UsersController', () => {
           },
         });
       });
+      it('returns 0 for user without any points', async () => {
```
Comment (NullSoldier) [inline src/users/users.controller.spec.ts:352]: Rename this to returns null

File: src/users/users.controller.spec.ts [hunk 6]
```diff
@@ -349,6 +349,40 @@ describe('UsersController', () => {
...
+          });
+
+        expect(body).toMatchObject({
+          user_id: user.id,
+          granularity: MetricsGranularity.LIFETIME,
+          points: 0,
+          node_uptime: {
+            total_hours: 0,
+            last_checked_in: null,
+          },
```
Comment (NullSoldier) [inline src/users/users.controller.spec.ts:371]: you could delete this since it is outside the scope of the test

**Comment by NullSoldier** [comment]
Then we'll also need to upgrade getRank() as well.

**Review by NullSoldier — APPROVED** [review]
